src/audio/level_multiplier/level_multiplier-ipc3.c

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2025 Intel Corporation.
+
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/component.h>
+#include "level_multiplier.h"
+
+LOG_MODULE_DECLARE(level_multiplier, CONFIG_SOF_LOG_LEVEL);
+
+/* This function handles the real-time controls. The ALSA controls have the
+ * param_id set to indicate the control type. The control ID, from topology,
+ * is used to separate the controls instances of same type. In control payload
+ * the num_elems defines to how many channels the control is applied to.
+ */
+__cold int level_multiplier_set_config(struct processing_module *mod, uint32_t param_id,
```

Comment:
I don't think anybody is using `__cold` with IPC3, but it probably doesn't hurt either. It's only defined to something non-trivial on ACE, and ACE doesn't use IPC3